File: test_ground/ts_ground/src/fns/case1.ts
export function case1() {
  const a = { num: 1 };
  const b = a;
  let c = a;

  a.num = 2;
  b.num = 4;
  c.num = 5;

  console.log(a);
  console.log(b);
  console.log(c);

  c = { num: 10 };

  console.log(c);
}

File: nsc/test_ground/rust_ground/src/fns/case1.rs
use std::{cell::RefCell, rc::Rc};
#[derive(Debug)]
struct _Ag1 {
  pub num: f64
}

pub fn case1() {
  let a = Rc::new(RefCell::new( _Ag1{ num: 1.0 }));
  let b = a.clone();
  let c = a.clone();

  a.borrow_mut().num = 2.0;
  b.borrow_mut().num = 4.0;
  c.borrow_mut().num = 5.0;

  println!("{:#?}", &a);
  println!("{:#?}", &b);
  println!("{:#?}", &c);

  *c.borrow_mut() = _Ag1{ num: 10.0 };

  println!("{:#?}", c);
}